tools/kvmexit.py

```diff
@@ -0,0 +1,393 @@
+#!/usr/bin/env python
+#
+# kvmexit.py
+#
+# Display the exit_reason and its statistics of each vm exit
+# for all vcpus of all virtual machines. For example:
+# $./kvmexit.py
+#  PID      TID      KVM_EXIT_REASON                     COUNT
+#  1273551  1273568  EXIT_REASON_MSR_WRITE               6
+#  1274253  1274261  EXIT_REASON_EXTERNAL_INTERRUPT      1
+#  1274253  1274261  EXIT_REASON_HLT                     12
+#  ...
+#
+# Besides, we also allow users to specify one pid, tid(s), or one
+# pid and its vcpu. See kvmexit_example.txt for more examples.
+#
+# @PID: each vitual machine's pid in the user space.
+# @TID: the user space's thread of each vcpu of that virtual machine.
+# @KVM_EXIT_REASON: the reason why the vm exits.
+# @COUNT: the counts of the @KVM_EXIT_REASONS.
+#
+# REQUIRES: Linux 4.7+ (BPF_PROG_TYPE_TRACEPOINT support)
```

Comment:
I have
```
$ git describe a38d1107
v4.20-rc6-1353-ga38d1107f937
```
So looks like we need 4.20+.

Reply:
> I have
> 
> ```
> $ git describe a38d1107
> v4.20-rc6-1353-ga38d1107f937
> ```
> 
> So looks like we need 4.20+.

When this tool is used by users, it will try to firstly employ raw tracepoints in modules, if fails then fall back to regular tracepoint. Thus when referring to 'REQUIRES', I think we'd better mention the time when regular tracepoint is introduced, instead of the time when raw tp is introduced. (The commit: a38d1107 is introduced to support raw tracepoints in modules) :)

Reply:
Okay. 4.7+ is fine then.